stored_procedures/create_run_record.sql

```diff
@@ -0,0 +1,84 @@
+-- drop the stored procedure
+DROP PROCEDURE IF EXISTS `biosero_uat`.`createRunRecord`;
+
+-- create the stored procedure
+DELIMITER $$
+
+-- Creates a run record row at the start of a workcell run on the automation_system_runs table
+-- and a linked row on the configurations table
+CREATE PROCEDURE `biosero_uat`.`createRunRecord` (
+  IN input_automation_system_name VARCHAR(255) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci,
+  IN input_system_run_id INT,
+  IN input_gbg_method_name VARCHAR(255) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci,
+  IN input_user_id VARCHAR(255) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci
+)
+BEGIN
+  -- Rollback if there is any error
+  DECLARE EXIT HANDLER FOR SQLEXCEPTION
+    BEGIN
+      SHOW ERRORS;
+      ROLLBACK;
+    END;
```

Comment:
Do we want to add this rollback to the other stored procedures too?

Reply:
I was thinking about the same thing but I've checked and the other stored procedures are one single insert, and normally they will be run in a batch (set well properties) while this one will be run alone.  The reason to add the transaction and rollback here was because there are 2 inserts and we want both or none applied. In the other stored procedures there is only one insert, so if it fails there is nothing to rollback. Anyway they will need a transaction in code, to group related wells together.